extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -43,50 +53,92 @@ public void handle(RoutingContext event) {
                 }
                 //we put the authenticator into the routing context so it can be used by other systems
                 event.put(HttpAuthenticator.class.getName(), authenticator);
-                authenticator.attemptAuthentication(event)
-                        .subscribe().with(new Consumer<SecurityIdentity>() {
-                            @Override
-                            public void accept(SecurityIdentity identity) {
-                                if (event.response().ended()) {
-                                    return;
+
+                //register the default auth failure handler
+                //if proactive auth is used this is the only one
+                //if using lazy auth this can be modified downstream, to control authentication behaviour
```

Comment:
Was just about to ask what it means and then saw `AUTH_FAILURE_HANDLER` docs :-)